File: careconnect2025/backend/core/src/main/java/com/careconnect/dto/PostWithCommentCountDto.java
package com.careconnect.dto;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor

public class PostWithCommentCountDto {
    private Long id;
    private Long userId;
    private String content;
    private String imageUrl;
    private LocalDateTime createdAt;
    private int commentCount;
    private String username;
}

File: careconnect2025/frontend/lib/features/social/presentation/model/PostWithCommentCountDto.dart
class PostWithCommentCountDto {
  final int id;
  final int userId;
  final String content;
  final String? imageUrl;
  final DateTime createdAt;
  final int commentCount;
  final String username;

  PostWithCommentCountDto({
    required this.id,
    required this.userId,
    required this.content,
    this.imageUrl,
    required this.createdAt,
    required this.commentCount,
    required this.username,
  });

  factory PostWithCommentCountDto.fromJson(Map<String, dynamic> json) {
    return PostWithCommentCountDto(
      id: json['id'],
      userId: json['userId'],
      content: json['content'],
      imageUrl: json['imageUrl'],
      createdAt: DateTime.parse(json['createdAt']),
      commentCount: json['commentCount'],
      username: json['username'] ?? 'Unknown User',
    );
  }
}